shopfloor/actions/change_package_lot.py

```diff
@@ -118,50 +106,45 @@ def change_package(self, move_line, package, response_ok_func, response_error_fu
             # a mistake in the data... fix the quant to move the package here
             inventory.move_package_quants_to_location(package, move_line.location_id)
 
-        # search a package level which would already move the scanned package
-        reserved_level = (
-            self.env["stock.package_level"].search([("package_id", "=", package.id)])
-            # not possible to search on state
-            .filtered(lambda level: level.state in ("new", "assigned"))
+        # search other move lines which should already pick the scanned package
+        other_reserved_lines = self.env["stock.move.line"].search(
+            [
+                ("package_id", "=", package.id),
+                ("state", "in", ("partially_available", "assigned")),
+            ]
+        )
+        # we can't change already picked lines
+        unreservable_lines = other_reserved_lines.filtered(
+            lambda line: line.qty_done == 0
         )
-        if reserved_level:
-            reserved_level.ensure_one()
-        if reserved_level.is_done:
-            # Not really supposed to happen: if someone sets is_done, the package
-            # should no longer be here! But we have to check this and inform the
-            # user in any case.
+        to_assign_moves = unreservable_lines.move_id
+        # /!\ be sure to box the side-effects before calling "replace_package"
+        # in the savepoint, as we catch the error, we must be sure that any
+        # change is rollbacked
+        try:
+            with self.env.cr.savepoint():
+                # if we leave the package level, it will try to reserve the same
+                # one again
+                unreservable_lines.package_level_id.explode_package()
+                # unreserve qties of other lines
+                unreservable_lines.unlink()
```

Comment:
I should maybe move the "unreservable" lines management in `replace_package` as well, as I'm not sure calling `replace_package` without doing this stuff can work.